src/server.cpp

```diff
@@ -40,23 +41,31 @@ void Server::asyncSend(const Packet& response, const std::function<void(const er
        [this, callback](const error_code& ec, std::size_t /*bytesTransferred*/) {handleSend(ec, callback);});
 }
 
-void Server::handleReceive(const error_code& error, std::size_t bytes, std::function<void(const error_code&, const Packet&)> callback)
+void Server::handleReceive(const error_code& error, std::size_t bytes, std::function<void(const error_code&, const std::optional<Packet>&)> callback)
 {
-    try
-    {
-        if (error)
-            throw std::runtime_error{"Error async_receivw_from: " + error.message()};
+        std::optional<Packet> oEmpty;
+        std::optional<Packet> oRequest;
```

Comment:
You do not need this variable, use `std::make_optional` where you pass the packet into the callback.